### ISSUE TITLE
[Bug]: Where did the 'Save' and 'Save all' buttons go???

### ISSUE THREAD

**Issue by wittend** [eranif/codelite#3151]
### What happened?

I consider this a serious accessibility and usability bug in Codelite !7 on Linux (and perhaps elsewhere).

Where did the 'Save' and 'Save all' buttons go???  Is there a way to put them back?  

I know that this is 'Old guy heresy' but one of he biggest reasons that I use Codelite is that you haven't fallen for the VS Code stupidity that a button to selectively save may work is to much for present-generation programmers to comprehend.  And no, I can't use Ctrl-S or other key sequences.  That is why I don't use EMACS.

Please, I implore you, let a usable toolbar be an option!  Don't dictate to the users!

Dave

### Version

Self compiled

### Operating system

Linux

### Steps to reproduce

```text
Just run the program.
```

**Comment by eranif** [comment]
I played with it, but I thought that I restore them :) 
If you can specify your OS, I might release an urgent fix

**Comment by eranif** [comment]
FYI: I did notice the "Linux", but I need the exact distro

**Comment by wittend** [comment]
Eran,

Thanks!

Excellent editor, as always!

I'm using Linux Mint Cinnamon Amd64 generic with Linux Jammy underneath.
Looks like I refresh from:
deb https://repos.codelite.org/ubuntu/ jammy universe
deb https://repos.codelite.org/wx3.2.0/ubuntu/ jammy universe

Thanks if you can do it.  I just don't want them to go away forever.  It
could be an option.

Dave

**Comment by eranif** [comment]
This means that I can upload deb for jammy and it will work  for you. good.
I will fix this by tomorrow probably

**Comment by wittend** [comment]
Amazing.  Thanks!

Dave

**Comment by wittend** [comment]
Awesome, thanks!

**Comment by eranif** [comment]
This deb should restore the save/save-all buttons:

```bash
wget https://codelite.org/downloads/codelite/wip/CodeLite-17.2.0-gtk3-ubuntu-jammy-x86_64.deb
sudo gdebi -i CodeLite-17.2.0-gtk3-ubuntu-jammy-x86_64.deb
```

**Comment by wittend** [comment]
Irfan,

Thanks so much!

Dave

**Comment by wittend** [comment]
Eranif,

Thank you so much.
A couple of issues though.

1) minor:  The gdebi command (at least on Mint) doesn't take a -i
argument.
2) so I decided that you meant to use 'dpkg -i'
3) that worked but reported package conflicts (gtk3 ?)
The updated program appeared to work well - as advertised.

4) I went and 'fixed broken packages' using Synaptic.
That seems to have disabled Codelite.

Attempt to reinstall using dpkg:

tmp$ sudo dpkg -i  CodeLite-17.2.0-gtk3-ubuntu-jammy-x86_64.deb
[sudo] password for dave:
Selecting previously unselected package codelite.
(Reading database ... 684159 files and directories currently installed.)
Preparing to unpack CodeLite-17.2.0-gtk3-ubuntu-jammy-x86_64.deb ...
Unpacking codelite (17.2.0-gtk3) ...
dpkg: dependency problems prevent configuration of codelite:
 codelite depends on clangd (>= 10); however:
  Package clangd is not installed.

dpkg: error processing package codelite (--install):
 dependency problems - leaving unconfigured
Processing triggers for gnome-menus (3.36.0-1ubuntu3) ...
Processing triggers for desktop-file-utils (0.26+mint1+vanessa) ...
Processing triggers for mailcap (3.70+nmu1ubuntu1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Errors were encountered while processing:
 codelite

Any suggestions?
(Seems like gtk3 has created problems in the past)

System:
  Kernel: 5.15.0-60-generic x86_64 bits: 64 compiler: gcc v: 11.3.0
Desktop: Cinnamon 5.4.12
    tk: GTK 3.24.33 wm: Mutter dm: LightDM Distro: Linux Mint 21 Vanessa
base: Ubuntu 22.04 jammy
Machine:
  Type: Desktop System: Gigabyte product: B550I AORUS PRO AX v: N/A serial:
<superuser required>
  Mobo: Gigabyte model: B550I AORUS PRO AX v: x.x serial: <superuser
required>
    UEFI: American Megatrends LLC. v: F15 date: 01/04/2022
Battery:
  Device-1: hidpp_battery_0 model: Logitech Wireless Keyboard serial:
<filter>
    charge: 55% (should be ignored) status: Discharging
CPU:
  Info: 6-core model: AMD Ryzen 5 5600G with Radeon Graphics bits: 64 type:
MT MCP arch: Zen 3
    rev: 0 cache: L1: 384 KiB L2: 3 MiB L3: 16 MiB
  Speed (MHz): avg: 3235 high: 3798 min/max: 1400/4464 boost: enabled
cores: 1: 3479 2: 2838
    3: 2864 4: 3798 5: 3224 6: 3094 7: 3384 8: 3114 9: 3330 10: 3786 11:
2817 12: 3099
    bogomips: 93423
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: AMD Cezanne vendor: Gigabyte driver: amdgpu v: kernel pcie:
speed: 8 GT/s lanes: 16
    ports: active: HDMI-A-1,HDMI-A-2 empty: DP-1 bus-ID: 07:00.0 chip-ID:
1002:1638
  Display: x11 server: X.Org v: 1.21.1.3 driver: X: loaded: amdgpu,ati
    unloaded: fbdev,modesetting,vesa gpu: amdgpu display-ID: :0 screens: 1
  Screen-1: 0 s-res: 4480x1440 s-dpi: 96
  Monitor-1: HDMI-A-0 mapped: HDMI-A-1 pos: primary,top-left model: Acer
ET322QU res: 2560x1440
    dpi: 93 diag: 801mm (31.5")
  Monitor-2: HDMI-A-1 mapped: HDMI-A-2 pos: primary,bottom-r model: AOC
24B2W1 res: 1920x1080
    dpi: 93 diag: 604mm (23.8")
  OpenGL: renderer: RENOIR (renoir LLVM 15.0.6 DRM 3.42 5.15.0-60-generic)
v: 4.6 Mesa 22.2.5
    direct render: Yes
Audio:
  Device-1: AMD Renoir Radeon High Definition Audio driver: snd_hda_intel
v: kernel pcie:
    speed: 8 GT/s lanes: 16 bus-ID: 07:00.1 chip-ID: 1002:1637
  Device-2: AMD Family 17h HD Audio vendor: Gigabyte driver: snd_hda_intel
v: kernel pcie:
    speed: 8 GT/s lanes: 16 bus-ID: 07:00.6 chip-ID: 1022:15e3
  Sound Server-1: ALSA v: k5.15.0-60-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: no
  Sound Server-3: PipeWire v: 0.3.63 running: yes
Network:
  Device-1: Realtek RTL8125 2.5GbE vendor: Gigabyte driver: r8169 v: kernel
pcie: speed: 5 GT/s
    lanes: 1 port: f000 bus-ID: 04:00.0 chip-ID: 10ec:8125
  IF: enp4s0 state: up speed: 1000 Mbps duplex: full mac: <filter>
  Device-2: MEDIATEK driver: mt7921e v: kernel pcie: speed: 5 GT/s lanes: 1
port: N/A
    bus-ID: 05:00.0 chip-ID: 14c3:0608
  IF: wlp5s0 state: up mac: <filter>
Bluetooth:
  Device-1: MediaTek Wireless_Device type: USB driver: btusb v: 0.8 bus-ID:
1-8:6
    chip-ID: 0e8d:0608
  Report: hciconfig ID: hci0 rfk-id: 0 state: up address: <filter> bt-v:
3.0 lmp-v: 5.2
    sub-v: 2202
RAID:
  Device-1: md0 type: mdraid level: mirror status: active size: 465.64 GiB
  Info: report: 2/2 UU blocks: 488254464 chunk-size: N/A super-blocks: 1.2
  Components: Online: 0: sdb 1: sdc
Drives:
  Local Storage: total: 3.76 TiB used: 688.46 GiB (17.9%)
  ID-1: /dev/nvme0n1 vendor: Samsung model: SSD 970 EVO Plus 1TB size:
931.51 GiB
    speed: 31.6 Gb/s lanes: 4 serial: <filter> temp: 36.9 C
  ID-2: /dev/nvme1n1 model: PCIe SSD size: 119.24 GiB speed: 15.8 Gb/s
lanes: 2 serial: <filter>
    temp: 40.9 C
  ID-3: /dev/sda vendor: Western Digital model: WD2003FZEX-00SRLA0 size:
1.82 TiB
    speed: 6.0 Gb/s serial: <filter>
  ID-4: /dev/sdb type: USB vendor: Samsung model: SSD 850 EVO 500GB size:
465.76 GiB
    serial: <filter>
  ID-5: /dev/sdc type: USB vendor: Samsung model: SSD 850 EVO 500GB size:
465.76 GiB
    serial: <filter>
Partition:
  ID-1: / size: 116.32 GiB used: 35.76 GiB (30.7%) fs: ext4 dev:
/dev/nvme1n1p3
  ID-2: /boot/efi size: 512 MiB used: 6.1 MiB (1.2%) fs: vfat dev:
/dev/nvme1n1p2
Swap:
  ID-1: swap-1 type: file size: 2 GiB used: 0 KiB (0.0%) priority: -2 file:
/swapfile
Sensors:
  System Temperatures: cpu: 34.0 C mobo: 31.0 C gpu: amdgpu temp: 28.0 C
  Fan Speeds (RPM): N/A
Repos:
  Packages: 4447 note: see --pkg apt: 4420 flatpak: 15 snap: 12
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/additional-repositories.list
    1: deb https: //repos.codelite.org/ubuntu/ jammy universe
    2: deb https: //repos.codelite.org/wx3.2.0/ubuntu/ jammy universe
    3: deb https: //repos.codelite.org/wx3.2/ubuntu/ jammy universe
  Active apt repos in:
/etc/apt/sources.list.d/appimagelauncher-team-stable-jammy.list
    1: deb http: //ppa.launchpad.net/appimagelauncher-team/stable/ubuntu
jammy main
  Active apt repos in: /etc/apt/sources.list.d/git-core-ppa-jammy.list
    1: deb http: //ppa.launchpad.net/git-core/ppa/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/google-chrome.list
    1: deb [arch=amd64] https: //dl.google.com/linux/chrome/deb/ stable main
  Active apt repos in: /etc/apt/sources.list.d/google-earth-pro.list
    1: deb [arch=amd64] http: //dl.google.com/linux/earth/deb/ stable main
  Active apt repos in:
/etc/apt/sources.list.d/kicad-kicad-6_0-releases-jammy.list
    1: deb http: //ppa.launchpad.net/kicad/kicad-6.0-releases/ubuntu jammy
main
  Active apt repos in:
/etc/apt/sources.list.d/official-package-repositories.list
    1: deb https: //plug-mirror.rcac.purdue.edu/mint vanessa main upstream
import backport
    2: deb http: //ftp.ussg.iu.edu/linux/ubuntu jammy main restricted
universe multiverse
    3: deb http: //ftp.ussg.iu.edu/linux/ubuntu jammy-updates main
restricted universe multiverse
    4: deb http: //ftp.ussg.iu.edu/linux/ubuntu jammy-backports main
restricted universe multiverse
    5: deb http: //security.ubuntu.com/ubuntu/ jammy-security main
restricted universe multiverse
  Active apt repos in:
/etc/apt/sources.list.d/pipewire-debian-pipewire-upstream-jammy.list
    1: deb http: //
ppa.launchpad.net/pipewire-debian/pipewire-upstream/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/signal-xenial.list
    1: deb [arch=amd64
signed-by=/usr/share/keyrings/signal-desktop-keyring.gpg] https: //
updates.signal.org/desktop/apt xenial main
  Active apt repos in: /etc/apt/sources.list.d/vscode.list
    1: deb [arch=amd64,arm64,armhf] http: //
packages.microsoft.com/repos/code stable main
Info:
  Processes: 399 Uptime: 8m Memory: 14.98 GiB used: 3.5 GiB (23.3%) Init:
systemd v: 249
  runlevel: 5 Compilers: gcc: 11.3.0 alt: 11/12 clang: 14.0.0-1ubuntu1
Client: Cinnamon v: 5.4.12
  inxi: 3.3.13

Dave
